lib/bugsnag.js

```diff
@@ -12,13 +12,16 @@ var domain = require("domain"),
 // Ensure we get all stack frames from thrown errors.
 Error.stackTraceLimit = Infinity;
 
+// Will ensure errors still logged if shouldNotify is false
 function autoNotifyCallback(notifiedError, uncaughtError) {
     if (!uncaughtError) {
         uncaughtError = notifiedError.domain;
     }
     return function(error) {
         if (error) {
-            Configuration.logger.error("Bugsnag: error notifying bugsnag.com - " + error);
+            Configuration.logger.error(Bugsnag.shouldNotify() ?
+                "error notifying bugsnag.com - " + error + error.stack :
+                "Bugsnag will not be notified - " + notifiedError + notifiedError.stack);
```

Comment:
`notifiedError` will be coerced into a string, which will contain the `Error: <message>` bit, but then the string `notifiedError.stack` also contains `Error: <message>` followed by each stack line.

In summary you only need `.stack` and not both. This goes for line 23 and 24.

I've had a tinker and coming at it from the user's perspective, I think the best output would be something like this…

```
Bugsnag:  configuration prevents the following error from being sent to Bugsnag.
https://docs.bugsnag.com/platforms/nodejs/other/configuration-options/#notifyreleasestages

Error: Some error occured
    at app.use (/Users/bengourley/Development/bugsnag-bug-repro/index.js:13:11)
    at dispatch (/Users/bengourley/Development/bugsnag-bug-repro/node_modules/koa-compose/index.js:42:32)
    at /Users/bengourley/Development/bugsnag-bug-repro/node_modules/koa-compose/index.js:34:12
    at Server.handleRequest (/Users/bengourley/Development/bugsnag-bug-repro/node_modules/koa/lib/application.js:136:14)
    at emitTwo (events.js:125:13)
    at Server.emit (events.js:213:7)
    at parserOnIncoming (_http_server.js:602:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:116:23)
```

Which you should be able to do with something like…

```js
[
  "configuration prevents the following error from being sent to Bugsnag.",
  "https://docs.bugsnag.com/platforms/nodejs/other/configuration-options/#notifyreleasestages",
  "",
  err.stack
].join('\n')
```